src/main/java/org/radarcns/management/service/RoleService.java

```diff
@@ -64,18 +68,23 @@ public RoleDTO save(RoleDTO roleDTO) {
             return result;
         }
         List<String> currentUserAuthorities = currentUser.getAuthorities().stream()
-                .map(Authority::getName).collect(
-                        Collectors.toList());
+                .map(Authority::getName).collect(Collectors.toList());
         if (currentUserAuthorities.contains(AuthoritiesConstants.SYS_ADMIN)) {
             log.debug("Request to get all Roles");
             result = roleRepository.findAll().stream()
                     .map(roleMapper::roleToRoleDTO)
                     .collect(Collectors.toCollection(LinkedList::new));
         } else if (currentUserAuthorities.contains(AuthoritiesConstants.PROJECT_ADMIN)) {
             log.debug("Request to get project admin's project Projects");
-//            result =  roleRepository.findAllRolesByProjectId(currentUser.getProject().getId()).stream()
-//                .map(roleMapper::roleToRoleDTO)
-//                .collect(Collectors.toCollection(LinkedList::new));
+            result = currentUser.getRoles().stream()
```

Comment:
What is the use-case of this implementation?

Reply:
Good question, i noticed the project admin would not be able to get a list of roles so i fixed it and moved on. We used to have distinct pages to view roles, the pages still exist but the links inside the application do not, since looking at a role by itself is not very useful.

From what i can tell right now, it does not seem this method is even necessary anymore. I will investigate a bit further and create a separate issue for removing unused code related to roles.